url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/ui/task/DefaultTaskRenderer.java

```diff
@@ -0,0 +1,379 @@
+package org.swiftgantt.ui.task;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Polygon;
+import java.awt.Rectangle;
+import java.awt.Shape;
+import java.util.Collection;
+import java.util.List;
+
+import javax.swing.JComponent;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+
+import org.swiftgantt.GanttChart;
+import org.swiftgantt.model.GanttModel;
+import org.swiftgantt.model.Task;
+import org.swiftgantt.model.TaskHelper;
+import org.swiftgantt.ui.PaintLogger;
+import org.swiftgantt.ui.TimeUnit;
+import org.swiftgantt.ui.timeaxis.TimeAxisUtils;
+
+/**
+ * The default implementation for {@link BaseTaskRenderer}.
+ * 
+ * @author Yuxing Wang
+ * @version 1.0
+ */
+public class DefaultTaskRenderer extends BaseTaskRenderer {
+
+	private int connectorStyle = 1;// 1 or 2.
+	private List<TaskWrapper> tasks = null; // Tasks that need to be paint.
+
+	public DefaultTaskRenderer(GanttChart gantt) {
+		super.gantt = gantt;
+		super.config = gantt.getConfig();
+	}
+
+	/**
+	 * @param rec The client area that paint the tasks.
+	 */
+	@Override
+	public void paint(Graphics g, JComponent c, Rectangle rec) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 45-51 into a separate method.